Sources/NIOCrashTester/CrashTests+ByteBuffer.swift

```diff
@@ -21,4 +21,10 @@ struct ByteBufferCrashTests {
         var buffer = ByteBufferAllocator().buffer(capacity: 16)
         buffer.moveReaderIndex(forwardBy: 1)
     }
+
+    let testAllocatingNegativeSize = CrashTest(
+        regex: #"^Precondition failed: ByteBuffer capacity must be positive."#
+    ) {
+        _ = ByteBufferAllocator().buffer(capacity: -1)
```

Comment:
Maybe I missed it but should we also have a public `ByteBuffer(capacity:)`? (or `minimumCapacity`)

Reply:
@glbrntt I didn't add those because `Array` and friends don't have that either. If you really want this, you can have it this way:

```
var buffer = ByteBuffer()
buffer.reserveCapacity(17)
```

like for `Array`.

Reply:
Fine by my, just wanted to check it wasn't an oversight

Reply:
I'm open to adding it. @Lukasa ?

Reply:
Nah, I think matching `Array` is fine. Leave it as-is.

Reply:
ok